apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,288 @@
+      return true;
+    }
+
+    if (state.ncesSchoolId && state.ncesSchoolId !== '-1') {
+      return true;
+    }
+
+    return !!(state.country && state.schoolType && state.schoolName);
+  }
+
+  componentDidMount() {
+    this.logEvent(FIREHOSE_EVENTS.SHOW);
+  }
+
+  buildSchoolData() {
+    const {country, schoolType, ncesSchoolId} = this.state;
+    // If we have an NCES id, _only_ send that - everything else will be
+    // backfilled by records on the server.
+    if (ncesSchoolId && ncesSchoolId !== '-1') {
+      return {
+        "user[school_info_attributes][school_id]": ncesSchoolId,
+      };
+    }
+
+    // If we don't know enough to pick other metadata, only send these.
+    if (!country || !schoolType) {
+      return {
+        "user[school_info_attributes][country]": country,
+        "user[school_info_attributes][school_type]": schoolType,
+      };
+    }
+
+    // If an NCES type is selected but we don't know anything else, send a
+    // blank NCES id to ensure we save the current input state.
+    const isUS = country === 'United States';
+    const isNcesSchoolType = isUS && SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(schoolType);
+    if (isNcesSchoolType && ncesSchoolId === '') {
+      return {
+        "user[school_info_attributes][country]": country,
+        "user[school_info_attributes][school_type]": schoolType,
+        "user[school_info_attributes][school_id]": ncesSchoolId,
+      };
+    }
+
+    return {
+      "user[school_info_attributes][country]": country,
+      "user[school_info_attributes][school_type]": schoolType,
+      "user[school_info_attributes][school_name]": this.state.schoolName,
+      "user[school_info_attributes][full_address]": this.state.schoolLocation,
+    };
+  }
+
+  handleSchoolInfoSubmit = () => {
+    this.logEvent(FIREHOSE_EVENTS.SUBMIT, {
+      attempt: this.state.showSchoolInfoUnknownError ? 2 : 1
+    });
+
+    const schoolData = this.buildSchoolData();
+    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
+    $.post({
+      url: formUrl + '.json',
+      dataType: "json",
+      data: {
+        '_method': 'patch',
+        [authTokenName]: authTokenValue,
+        ...schoolData,
+      },
+    }).done(() => {
+      this.logEvent(FIREHOSE_EVENTS.SAVE_SUCCESS, {
+        attempt: this.state.showSchoolInfoUnknownError ? 2 : 1
+      });
+
+      this.props.onClose();
+    }).fail(() => {
+      this.logEvent(FIREHOSE_EVENTS.SAVE_FAILURE, {
+        attempt: this.state.showSchoolInfoUnknownError ? 2 : 1
+      });
+
+      if (!this.state.showSchoolInfoUnknownError) {
+        // First failure, display error message and give the teacher a chance
+        // to try again.
+        this.setState({showSchoolInfoUnknownError: true});
+      } else {
+        // We already failed once, let's not block the teacher any longer.
+        // TODO (bbuchanan): Gather metrics on this.
```

Comment:
Did you intend to handle this TODO now or just eventually?

Reply:
This is handled, I just forgot to remove the TODO.  Thanks!